File: week2/readability/readability.c
//CS50 Readability -  Owen FitzGerald June 2021

#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>

// Decalare fucntions
float count_number_of_sentences(string text);
float count_number_of_words(string text);
float count_number_of_letters(string text);

// Decalare global variables
float sentences;
float words;
int textLength;
float letters;

int main(void)
{
    // Get user input
    string text = get_string("Text: ");
    textLength = strlen(text);

    // Call functions
    count_number_of_letters(text);
    count_number_of_words(text);
    count_number_of_sentences(text);

    // Calculate averages and apply to Coleman-Liau formula
    float averageLetters = letters / words * 100;
    float averageSentences = sentences / words * 100;
    float colemanLiau = 0.0588 * averageLetters - 0.296 * averageSentences - 15.8;

    if (colemanLiau > 16)
    {
        printf("Grade 16+\n");
    }
    else if (colemanLiau < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %.0f\n", colemanLiau);
    }

    // printf("%f letter(s)\n", letters);
    // printf("%f word(s)\n", words);
    // printf("%f sentence(s)\n", sentences);
    // printf("String length: %i\n", textLength);
    // printf("%.2f average letters\n", averageLetters);
    // printf("%.2f average sentences\n", averageSentences);

}

float count_number_of_sentences(string text)
{
    sentences = 1;
    for (int i = 0; i < textLength; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            if (isblank(text[i + 1]))
            {
                sentences += 1;
            }
        }
    }
    return sentences;
}

float count_number_of_words(string text)
{
    words = 1;
    for (int i = 0; i < textLength; i++)
    {
        if (isblank(text[i]))
        {
            words += 1;
        }
    }
    return words;
}

float count_number_of_letters(string text)
{
    letters = 0;
    for (int i = 0; i < textLength; i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters += 1;
        }
    }
    return letters;
}

File: week6/pset6/readability.py
# //CS50 Readability - Owen FitzGerald 4/11/2021

from cs50 import get_string
from cs50 import get_int


def main():
    text = ask_user_for_text()
    global text_length
    text_length = len(text)

    # Call functions
    letters = count_number_of_letters(text)
    print(letters)

    words = count_number_of_words(text)
    print(words)

    sentences = count_number_of_sentences(text)
    print(sentences)

def count_number_of_letters(text):

    letters = 0
    for i in range(0, text_length, 1):
        if (text[i] >= 'a' and text[i] <= 'z') or (text[i] >= 'A' and text[i] <= 'Z'):
            letters += 1
    return(letters)


def count_number_of_words(text):

    words = 1
    for i in range(0, text_length, 1):
        if text[i] == ' ':
            words += 1
    return(words)


def count_number_of_sentences(text):

    sentences = 0
    for i in range(0, text_length, 1):
        if text[i] == '.' or text[i] == '?' or text[i] == '!':
            #if text[i +1] == ' ':
            sentences += 1
    return(sentences)



def ask_user_for_text():
    text = input("Text: ")
    return (text)

# standard way to execute main
if __name__ == "__main__":
    main()